sdk/keyvault/azure-security-keyvault-certificates/src/test/java/com/azure/security/keyvault/certificates/CertificateAsyncClientTest.java

```diff
@@ -0,0 +1,1035 @@
+
+        getCertificateIssuerRunner((issuer) -> {
+            CertificateIssuer createdIssuer = certificateAsyncClient.createIssuer(issuer).block();
+            StepVerifier.create(certificateAsyncClient.getIssuer(issuer.getName()))
+                    .assertNext(retrievedIssuer -> {
+                        assertTrue(issuerCreatedCorrectly(createdIssuer, retrievedIssuer));
+                    });
+        });
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getTestParameters")
+    public void getCertificateIssuerNotFound(HttpClient httpClient, CertificateServiceVersion serviceVersion) {
+        createCertificateAsyncClient(httpClient, serviceVersion);
+
+        StepVerifier.create(certificateAsyncClient.backupCertificate("non-existing"))
+            .verifyErrorSatisfies(e -> {
+                assertRestException(e, ResourceNotFoundException.class, HttpURLConnection.HTTP_NOT_FOUND);
+            });
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getTestParameters")
+    public void deleteCertificateIssuer(HttpClient httpClient, CertificateServiceVersion serviceVersion) {
+        createCertificateAsyncClient(httpClient, serviceVersion);
+
+        deleteCertificateIssuerRunner((issuer) -> {
+            CertificateIssuer createdIssuer = certificateAsyncClient.createIssuer(issuer).block();
+            StepVerifier.create(certificateAsyncClient.deleteIssuer(issuer.getName()))
+                    .assertNext(deletedIssuer -> {
+                        assertTrue(issuerCreatedCorrectly(createdIssuer, deletedIssuer));
+                    }).verifyComplete();
+        });
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getTestParameters")
+    public void deleteCertificateIssuerNotFound(HttpClient httpClient, CertificateServiceVersion serviceVersion) {
+        createCertificateAsyncClient(httpClient, serviceVersion);
+
+        StepVerifier.create(certificateAsyncClient.backupCertificate("non-existing"))
+            .verifyErrorSatisfies(e ->
+                assertRestException(e, ResourceNotFoundException.class, HttpURLConnection.HTTP_NOT_FOUND));
+    }
+
+    @ParameterizedTest(name = DISPLAY_NAME_WITH_ARGUMENTS)
+    @MethodSource("getTestParameters")
+    public void listCertificateIssuers(HttpClient httpClient, CertificateServiceVersion serviceVersion) {
+        createCertificateAsyncClient(httpClient, serviceVersion);
+
+        listCertificateIssuersRunner((certificateIssuers) -> {
+            HashMap<String, CertificateIssuer> certificateIssuersToList = new HashMap<>(certificateIssuers);
+
+            for (CertificateIssuer issuer : certificateIssuersToList.values()) {
+                CertificateIssuer certificateIssuer = certificateAsyncClient.createIssuer(issuer).block();
+
+                assertTrue(issuerCreatedCorrectly(issuer, certificateIssuer));
+            }
```

Comment:
If we follow Srikanta's recommendation, we should remove all blocking statements and uses of `SyncPoller`.

Reply:
Covered as part of this filed issue, https://github.com/Azure/azure-sdk-for-java/issues/31252